hoomd3/classic/butane/ann/butane_ANN.py

```diff
@@ -0,0 +1,320 @@
+    dihedral.params["CCCC"] = dict(k1=1.3, k2=-0.05, k3=0.2, k4=0.0)
+    dihedral.params["HCCC"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+    dihedral.params["HCCH"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+
+    lj_special_pairs = hoomd.md.special_pair.LJ()
+    lj_special_pairs.params["CCCC"] = dict(epsilon=0.07, sigma=3.55)
+    lj_special_pairs.params["HCCH"] = dict(epsilon=0.03, sigma=2.42)
+    lj_special_pairs.params["HCCC"] = dict(epsilon=sqrt(0.07 * 0.03), sigma=sqrt(3.55 * 2.42))
+    lj_special_pairs.r_cut["CCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs = hoomd.md.special_pair.Coulomb()
+    coulomb_special_pairs.params["CCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCH"] = dict(alpha=0.5)
+    coulomb_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs.r_cut["CCCC"] = 12.0
+    coulomb_special_pairs.r_cut["HCCC"] = 12.0
+    integrator = hoomd.md.Integrator(dt=dt)
+    integrator.forces.append(lj)
+    integrator.forces.append(coulomb[0])
+    integrator.forces.append(coulomb[1])
+    integrator.forces.append(harmonic)
+    integrator.forces.append(angle)
+    integrator.forces.append(dihedral)
+    integrator.forces.append(lj_special_pairs)
+    integrator.forces.append(coulomb_special_pairs)
+    nvt = hoomd.md.methods.Langevin(filter=hoomd.filter.All(), kT=kT)
+    integrator.methods.append(nvt)
+    sim.operations.integrator = integrator
+
+    return sim
+
+
+# %%
+def get_args(argv):
+    available_args = [
+        ("time-steps", "t", int, 5e5, "Number of simulation steps"),
+        # ("train-freq", "f", int, 5e3, "Frequency for neural network training"),
+    ]
+    parser = argparse.ArgumentParser(description="Example script to run ANN")
+    for name, short, T, val, doc in available_args:
+        parser.add_argument("--" + name, "-" + short, type=T, default=T(val), help=doc)
+
+    return parser.parse_args(argv)
+
+
+# %%
+def main(argv=[]):
+    args = get_args(argv)
+
+    cvs = [DihedralAngle([0, 4, 7, 10])]
+    grid = Grid(lower=(-pi,), upper=(pi,), shape=(64,), periodic=True)
+    topology = (8, 8)
+
+    method = ANN(cvs, grid, topology, kT)
+
+    tic = time.perf_counter()
+    raw_result = pysages.run(method, generate_context, args.time_steps)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
+
+    # analyze results and plot the free energy
+    result = pysages.analyze(raw_result)
+
+    mesh = result["mesh"]
+    A = result["free_energy"]
+
+    fig, ax = plt.subplots()
+
+    ax.set_xlabel(r"Dihedral Angle, $\xi$")
+    ax.set_ylabel(r"$A(\xi)$")
+
+    ax.plot(mesh, A)
+    plt.gca()
+    fig.savefig("butane-fe.png")
+
+    # write free energy to file
+    dih_vs_A = np.stack([mesh[:, 0], A[:, 0]], axis=1)
+    np.savetxt("butane-fe.dat", dih_vs_A, header='"Dihedral angle" "Free energy"')
```

Comment:
As discussed, let's switch to pickling (see this PR for an example: https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files)

Reply:
Done in [ANN] (https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/8e853d9b1e0b49375e2e36d24c97c987a48baabe)